File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        obj = {"(": ")", "[": "]", "{": "}"}
        arr = []
        length = len(s)
        k=obj.keys()
        for i in s:
            if i in k:
                arr.append(i)
            else:
                if len(arr)==0:
                    return False
                if i==obj[arr[-1]]:
                    arr.pop()
                else:
                    return False
        return len(arr)==0

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let obj={"(":")","[":"]","{":"}"}
    let arr=[];
    let len=s.length;
    for(let i=0;i<len;i++){
        if(s[i]=="(" || s[i]=="[" || s[i]=="{"){
            arr[arr.length]=s[i];
        }
        else{
            if (arr.length === 0){
                return false;
            }
            if(s[i]==obj[arr[arr.length-1]]){
                arr.pop();
            }
            else{
                return false;
            }
        }
    }
    if(arr.length===0){
        return true;
    }
    else{
        return false;
    }
};